### DISCUSSION TITLE
Add Gemini/Claude/Groq support without litellm

### DISCUSSION THREAD

**Discussion by jwu-sym** [open-webui/open-webui#2835] [General]
Please add Gemini/Claude/Groq support without litellm.
These three providers became very important for AI apps. In contrast, ollama models seemed less useful, maybe just llama3 and refined gguf.
Litellm had too many issues to be integral part of open-webui, imho.
OpenAI models are supported so well in open-webui, kudos.

**Comment by justinh-rahb** [comment]
We have a [Pipeline](https://github.com/open-webui/pipelines) for Anthropic: [anthropic_manifold_pipeline.py](https://github.com/open-webui/pipelines/blob/main/examples/pipelines/providers/anthropic_manifold_pipeline.py)

Groq you can just use as an "OpenAI" connection with this endpoint: `https://api.groq.com/openai/v1`

**Comment by jwu-sym** [comment]
Thank you @justinh-rahb for your suggestion.
I guess half of the functions are there. Groq "OpenAI" models and OpenAI 's has to be exclusive, doesn't it mean I can't configure both at the same time?

**Reply by justinh-rahb** [reply]
You can have multiple OpenAI connections configured. Use the `+` button to add another line and fill in the base URL and key for the service. I've got one stack with 6 or 7 different "OpenAI" endpoints.

**Comment by jwu-sym** [comment]
Great, thank you. problem mostly solved.

**Comment by dewzor** [comment]
Id really like this too. Right now id have loved to be able to install Open Webui on a small VPS and be able to use Groq API with it. Forgive my ignorance this might be possible right now, but afaik its not unless i dig into documentation deeply and customize the install.

**Reply by justinh-rahb** [reply]
Add it as an "OpenAI" connection in **Admin Settings > Connections** with base URL `https://api.groq.com/openai/v1` and your Groq key.

**Reply by dewzor** [reply]
awesome, thank you!
